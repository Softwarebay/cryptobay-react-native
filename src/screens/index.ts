export { HomeScreen } from './HomeScreen/HomeScreen';
export { WalthoughtScreen } from './WalthoughtScreen/WalthoughtScreen';
export { Screens } from './screens.enum';
