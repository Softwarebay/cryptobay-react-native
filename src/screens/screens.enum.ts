export enum Screens {
  HOME = 'Home',
  WALTHOUGHT = 'Walthought',
  SWAP = 'Swap',
  SETTINGS = 'Settings',
}
