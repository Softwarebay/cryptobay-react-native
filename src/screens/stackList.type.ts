export type StackList = {
  Home: undefined;
  Settings: undefined;
  Swap: undefined;
  WalletSetup: undefined;
  Walthought: undefined;
};
