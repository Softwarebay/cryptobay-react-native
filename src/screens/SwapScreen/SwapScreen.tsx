import { NativeStackScreenProps } from '@react-navigation/native-stack';
import React from 'react';
import { StyleSheet, View, SafeAreaView } from 'react-native';

import { Header } from '../../components/atoms';
import { Button } from '../../components/molecules';
import { Screens } from '../../screens.enum';
import { StackList } from '../../types/stackList.type';

type SwapScreenProps = NativeStackScreenProps<StackList, Screens.SWAP>;

export const SwapScreen: React.FC<SwapScreenProps> = ({
  navigation,
}: SwapScreenProps) => {
  const onPressBack = () => navigation.goBack();

  return (
    <SafeAreaView style={styles.safe}>
      <View style={styles.container}>
        <Header style={styles.header} value="Swap in progress" />
        <Button
          testID="swap-screen-back-button"
          label="Back"
          variant="ghost"
          onPress={onPressBack}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  safe: {
    width: '100%',
    height: '100%',
    paddingBottom: 30,
    flex: 1,
    backgroundColor: '#17171A',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  container: {
    width: '100%',
    marginTop: 300,
    alignItems: 'center',
  },
  header: {
    textAlign: 'center',
  },
});
