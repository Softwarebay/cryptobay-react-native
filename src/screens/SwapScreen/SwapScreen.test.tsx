import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import 'react-native';
import renderer from 'react-test-renderer';

import { Screens } from '../../screens.enum';

import { SwapScreen } from './SwapScreen';

const mockNavigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

describe('SwapScreen Screen', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <SwapScreen
          navigation={mockNavigation as any}
          route={Screens.SWAP as any}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure navigation works correctly', () => {
    render(
      <SwapScreen
        navigation={mockNavigation as any}
        route={Screens.SWAP as any}
      />,
    );
    fireEvent.press(screen.getByTestId('swap-screen-back-button'));
    expect(mockNavigation.goBack).toHaveBeenCalled();
  });
});
