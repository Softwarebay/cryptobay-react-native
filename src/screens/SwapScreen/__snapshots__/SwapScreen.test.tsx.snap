// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapScreen Screen Component Snapshot 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#17171A",
      "flex": 1,
      "height": "100%",
      "justifyContent": "space-between",
      "paddingBottom": 30,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 300,
        "width": "100%",
      }
    }
  >
    <RNCMaskedView>
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "lineHeight": 56,
              },
              Object {
                "fontSize": 40,
                "fontWeight": "400",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          Swap in progress
        </Text>
      </View>
      <BVLinearGradient
        colors={
          Array [
            4289318399,
            4285724369,
            4288736653,
            4294890341,
            4294616222,
          ]
        }
        endPoint={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
                "lineHeight": 56,
              },
              Object {
                "fontSize": 40,
                "fontWeight": "400",
              },
              Object {
                "opacity": 0,
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          Swap in progress
        </Text>
      </BVLinearGradient>
    </RNCMaskedView>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 8,
          "color": "#FEBF32",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
          "width": "100%",
        }
      }
      testID="swap-screen-back-button"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            Object {
              "color": "#FEBF32",
            },
          ]
        }
      >
        Back
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
